tests/model_registry/model_catalog/upgrade/test_catalog_upgrade.py

```diff
@@ -0,0 +1,429 @@
+import pytest
+import yaml
+from kubernetes.dynamic import DynamicClient
+
+from ocp_resources.config_map import ConfigMap
+from ocp_resources.deployment import Deployment
+from ocp_resources.pod import Pod
+from ocp_resources.route import Route
+from ocp_resources.service import Service
+
+from tests.model_registry.model_catalog.utils import (
+    validate_model_catalog_resource,
+    wait_for_model_catalog_api,
+    execute_get_command,
+    ResourceNotFoundError,
+    validate_default_catalog,
+)
+from tests.model_registry.model_catalog.constants import DEFAULT_CATALOG_ID
+from tests.model_registry.constants import DEFAULT_MODEL_CATALOG
+from simple_logger.logger import get_logger
+from timeout_sampler import retry
+
+from .constants import TEST_DATA
+
+LOGGER = get_logger(name=__name__)
+
+
+@pytest.mark.usefixtures(
+    "model_registry_namespace", "admin_client", "model_catalog_rest_url", "model_registry_rest_headers"
+)  # noqa: E501
+class TestPreUpgradeCatalog:
+    def _get_sources_response(self, model_catalog_rest_url: list[str], model_registry_rest_headers: dict[str, str]):
+        """Helper method to get sources response with retry logic"""
+
+        @retry(wait_timeout=120, sleep=10, exceptions_dict={AssertionError: [], ResourceNotFoundError: []})
+        def get_sources():
+            return execute_get_command(url=f"{model_catalog_rest_url[0]}sources", headers=model_registry_rest_headers)
+
+        return get_sources()
+
+    @pytest.mark.pre_upgrade
+    def test_create_custom_catalog_setup_pre_upgrade(
+        self,
+        custom_catalog_setup: dict,
+    ):
+        """Create and populate custom catalog source with test data (permanent modification for upgrade)"""
+        LOGGER.info("Starting custom catalog creation for upgrade testing")
+        # Fixture handles the setup automatically
+        LOGGER.info(f"Custom catalog setup completed: {custom_catalog_setup['catalog_id']}")
+
+        # Simple assertion to verify the fixture worked
+        assert custom_catalog_setup["catalog_id"] == TEST_DATA["catalog_id"]
+        assert custom_catalog_setup["models"] == TEST_DATA["models"]
```

Comment:
Not sure if we need this, since test_verify_custom_catalog_exists_pre_upgrade is ensuring what you expect is coming from the api calls.